airbyte-workers/src/main/java/io/airbyte/workers/process/DockerProcessBuilderFactory.java

```diff
@@ -53,6 +54,11 @@ public DockerProcessBuilderFactory(Path workspaceRoot, String workspaceMountSour
 
   @Override
   public ProcessBuilder create(final Path jobRoot, final String imageName, final String... args) {
+
+    if (!checkImageExists(imageName)) {
+      throw new IllegalStateException("Could not find image: " + imageName);
```

Comment:
this feels like it should be a named exception that gets caught by the worker and then sets a sensible message in the job output.